### ISSUE TITLE
Wrong IP address displayed

### ISSUE THREAD

**Issue by radiocolin** [CityOfPhiladelphia/property2#181]
When you look up property information by owner, this line is displayed at the top of the search results:

"Your IP address is 170.115.248.21 and you searched for properties owned by FOO on 11/26/2016 14:28."

The IP address displayed is not the client IP, it's the IP address of the phila.gov server.

**Comment by rmx90210** [comment]
I can't seem to reproduce this on my end. When I run an owner search from my phone I see a Verizon IP address. @radiocolin are you still seeing the issue?

**Comment by radiocolin** [comment]
Yes. Just did it on a Comcast connection and I get "Your IP address is
170.115.248.21 and you searched for properties owned by WEIR on
12/15/2016 10:57."


170.115.0.0/16 belongs to the city of Philadelphia.



-CW





On Thu, Dec 15, 2016, at 10:56 AM, Robert Martin wrote:

> I can't seem to reproduce this on my end. When I run an owner search
> from my phone I see a Verizon IP address. @radiocolin[1] are you still
> seeing the issue?
> — You are receiving this because you were mentioned. Reply to this
> email directly, view it on GitHub[2], or mute the thread[3].
> 




Links:

  1. https://github.com/radiocolin
  2. https://github.com/CityOfPhiladelphia/property2/issues/181#issuecomment-267363581

**Comment by rmx90210** [comment]
@radiocolin interesting. Could you paste in the URL for that search? It sounds like your request is getting proxied somehow, though I can't imagine what would be doing that. :thinking:

**Comment by radiocolin** [comment]
http://property.phila.gov/?o=weir



I spoke to someone who is midly in the know about this and they say
there is a reverse proxy/load balancer involved in the site, which is
likely the cause if that's the case.


-CW





On Thu, Dec 15, 2016, at 11:22 AM, Robert Martin wrote:

> @radiocolin[1] interesting. Could you paste in the URL for that
> search? It sounds like your request is getting proxied somehow, though
> I can't imagine what would be doing that. 🤔
> — You are receiving this because you were mentioned. Reply to this
> email directly, view it on GitHub[2], or mute the thread[3].
> 




Links:

  1. https://github.com/radiocolin
  2. https://github.com/CityOfPhiladelphia/property2/issues/181#issuecomment-267370904

**Comment by rmx90210** [comment]
@radiocolin I think that's true of the new phila.gov but Property is hosted here on GitHub, so the load balancer shouldn't be a factor AFAIK. When I hit http://property.phila.gov/?o=weir from the City network I do see the same IP you're getting, but from my phone I'm seeing a random cell network IP. And if I go on City wifi I see yet another IP. Sorry for the dumb question, but are you sure you're not on a City network somehow? Over VPN or something like that? Can't think of any other reason your client would be getting back a City IP. Really weird.

**Comment by radiocolin** [comment]
Hey, you're right.



I tried to do some more differential testing.



At home and work (all Comcast) I see the 170 city of Philadelphia IP
when I'm using Safari or Chrome on Mac OS.


If I visit on my T-Mobile iPhone, I get a T-Mobile IP address. If I turn
on wifi and connect to the Comcast  network, the correct Comcast IP
address shows up. So perhaps it's an OS issue?


Could it be that on my desktop I'm coming in over IPv6
(property.phila.gov has an AAAA record of "cityofphiladelphia.github.io.
1975 IN CNAME github.map.fastly.net.") and not on the phone?


-CW





On Thu, Dec 15, 2016, at 11:33 AM, Robert Martin wrote:

> @radiocolin[1] I think that's true of the new phila.gov but Property
> is hosted here on GitHub, so the load balancer shouldn't be a factor
> AFAIK. When I hit http://property.phila.gov/?o=weir from the City
> network I do see the same IP you're getting, but from my phone I'm
> seeing a random cell network IP. And if I go on City wifi I see yet
> another IP. Sorry for the dumb question, but are you sure you're not
> on a City network somehow? Over VPN or something like that? Can't
> think of any other reason your client would be getting back a City IP.
> — You are receiving this because you were mentioned. Reply to this
> email directly, view it on GitHub[2], or mute the thread[3].
> 




Links:

  1. https://github.com/radiocolin
  2. https://github.com/CityOfPhiladelphia/property2/issues/181#issuecomment-267374197
